src/Compilers/CSharp/Portable/Binder/BindingDiagnosticBag.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static class BindingDiagnosticBagExtensions
+    {
+        internal static CSDiagnosticInfo Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, ErrorCode code, Location location)
+        {
+            var info = new CSDiagnosticInfo(code);
+            diagnostics.DiagnosticBag?.Add(new CSDiagnostic(info, location));
+            return info;
+        }
+
+        internal static CSDiagnosticInfo Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, ErrorCode code, Location location, params object[] args)
+        {
+            var info = new CSDiagnosticInfo(code, args);
+            diagnostics.DiagnosticBag?.Add(new CSDiagnostic(info, location));
+            return info;
+        }
+
+        internal static CSDiagnosticInfo Add(this CodeAnalysis.BindingDiagnosticBag diagnostics, ErrorCode code, Location location, ImmutableArray<Symbol> symbols, params object[] args)
+        {
+            var info = new CSDiagnosticInfo(code, args, symbols, ImmutableArray<global::Microsoft.CodeAnalysis.Location>.Empty);
```

Comment:
>global:: [](start = 80, length = 8)

Unintentional `global::`?

Reply:
>Unintentional global::?

Not sure why IDE does that from time to time

---
In reply to: [361553800](https://github.com/dotnet/roslyn/pull/40560#discussion_r361553800) [](ancestors = 361553800)

Reply:
Removed in [https://github.com/dotnet/roslyn/pull/40640](https://github.com/dotnet/roslyn/pull/40640)

---
In reply to: [361554184](https://github.com/dotnet/roslyn/pull/40560#discussion_r361554184) [](ancestors = 361554184,361553800)